shinkai-bin/shinkai-node/tests/it/db_job_tests.rs

```diff
@@ -862,4 +868,48 @@ mod tests {
         let job_message_3: JobMessage = serde_json::from_str(&message_content_3).unwrap();
         assert_eq!(job_message_3.content, "Hello World 3".to_string());
     }
+
+    #[test]
+    fn test_add_forked_job() {
```

Comment:
messages can we like a tree:

            1
            ├── 2
            │   └── 4
            └── 3
            
 can we update the test so it reflects something like that?